src/main/java/com/greenbits/binlog/RowEventListenerAdapter.java

```diff
@@ -22,30 +20,32 @@ public RowEventListenerAdapter(RowEventListener listener,
                                    String databaseName) {
 
         this.listener = listener;
-        this.tableIdToTableMapEvent = new HashMap<Long, TableMapEvent>();
+        this.tableIdToTableMapEvent = new HashMap<Long, TableMapEventData>();
         this.checkpointer = checkpointer;
         this.databaseName = databaseName;
     }
 
     @Override
-    public void onEvents(BinlogEventV4 event) {
+    public void onEvent(Event event) {
+        EventData eventData = event.getData();
         try {
-            if (event instanceof FormatDescriptionEvent) {
-                handleFormatDescriptionEvent((FormatDescriptionEvent) event);
-            } else if (event instanceof WriteRowsEventV2) {
-                handleWriteRowsEvent((WriteRowsEventV2) event);
-            } else if (event instanceof UpdateRowsEventV2) {
-                handleUpdateRowsEvent((UpdateRowsEventV2) event);
-            } else if (event instanceof DeleteRowsEventV2) {
-                handleDeleteRowsEvent((DeleteRowsEventV2) event);
-            } else if (event instanceof RotateEvent) {
-                handleRotateEvent((RotateEvent) event);
-            } else if (event instanceof TableMapEvent) {
-                handleTableMapEvent((TableMapEvent) event);
-            } else if (event instanceof QueryEvent) {
-                handleQueryEvent((QueryEvent) event);
-            } else if (event instanceof XidEvent) {
-                handleXidEvent((XidEvent)event);
+            if (eventData instanceof FormatDescriptionEventData) {
+                handleFormatDescriptionEvent(event);
+            } else if (eventData instanceof WriteRowsEventData) {
+                handleWriteRowsEvent(event);
+            } else if (eventData instanceof UpdateRowsEventData) {
+                handleUpdateRowsEvent(event);
+            } else if (eventData instanceof DeleteRowsEventData) {
+                handleDeleteRowsEvent(event);
+            } else if (eventData instanceof RotateEventData) {
+                handleRotateEvent(event);
+            } else if (eventData instanceof TableMapEventData) {
+                handleTableMapEvent(event);
+            } else if (eventData instanceof QueryEventData) {
+                handleQueryEvent(event);
+            } else if (eventData instanceof XidEventData) {
+                handleXidEvent(event);
+            } else if (event.getHeader().getEventType() == EventType.ANONYMOUS_GTID){
```

Comment:
Just skipping it, not 100% sure what it is either, but I believe GTID is handled in a new format on this new library, but we don't use it regardless